DiskIOMeter.c

```diff
@@ -56,41 +55,50 @@ static void DiskIOMeter_updateValues(Meter* this) {
       }
 
       cached_last_update = host->realtimeMs;
+   }
 
-      if (status == RATESTATUS_NODATA) {
-         xSnprintf(this->txtBuffer, sizeof(this->txtBuffer), "no data");
-         return;
-      }
+   if (hasNewData) {
+      static uint64_t cached_read_total = 0;
```

Comment:
No need to set these variables to zero, the compiler is required to do this for static variables.

I know its not new with your code change here (though more static variables are added), but it would be better to move all these local static variables into the DiskIOMeter structure so they only take up space when that meter is in use (and dynamic space, rather than the global data segment).

Reply:
@natoscott
1. I would rather make the initialization explicit, for readability's sake. Not having an initial value implies me that the initial value is unimportant.
2. The commit like 1232e8274b5d584c6032d15c2af7ced212b9f300 ("Always update 'values' member of DiskIOMeter") is to address multiple instances of the meter. The cached values are class-wide, but the `values` members are per-instance, and the distinction has to be made, even if that mean a little waste of space. (Maybe we can introduce a `struct DiskIOMeterCache`) that allocates dynamically when in use, but I'm not working on it on this PR.)

```c
typedef struct {
   uint64_t lastUpdate;
   uint64_t totalBytesRead;
   uint64_t totalBytesWritten;
   uint64_t totalMsTimeSpent;
   uint64_t readDiff;
   uint64_t writeDiff;
   double utilisationDiff;
} DiskIOMeterCache;

static DiskIOMeterCache* cache = NULL;
```

Reply:
In terms of point 1, the initial value of a static is certainly not unimportant - its so important it has been part of the 
C language definition since day 1.  Please leave that code as-is.  Experienced C programmers know this sort of thing so there's no need to explicitly initialize to the already defined initial value.

For point 2, I agree - for now lets keep things static but a cache could be a neat addition IMO.  Again, no need to set the global cache variable to NULL - this is well defined and widely known.